Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -819,9 +818,9 @@ void PcapLiveDevice::setDefaultGateway()
 	if (retVal == NO_ERROR)
 	{
 		PIP_ADAPTER_INFO curAdapterInfo = adapterInfo;
+		std::string name(m_Name);

```

Comment:
I'm not sure why you moved this line to here

Reply:
I had placed this line out of scope of the loop because the value is not changed there.